core/types/calls.cc

```diff
@@ -957,10 +961,36 @@ DispatchResult dispatchCallSymbol(const GlobalState &gs, const DispatchArgs &arg
             } else {
                 // A keyword splat was passed in, but none of the declared arguments are keyword splats,
                 // or there is a keyword splat argument, but there is one ore more keyword non-splat argument
-                if (auto e = gs.beginError(args.callLoc(), errors::Infer::UntypedSplat)) {
-                    e.setHeader("Passing a hash where the specific keys are unknown to a method taking keyword "
-                                "arguments");
+                auto kwSplatArgLoc = args.argLoc(args.args.size() - 1);
```

Comment:
This PR adds tests which suggest that assuming the kwsplat arg is the last arg is not a correct assumption.

Reply:
It is last and the only arg, because we use Desugar to change `**opts, x: nil` into a single splat arg using `<Magic>.<to-hash-dup>` and `<Magic>.<merge>`